torch/csrc/dynamo/init.cpp

```diff
@@ -33,6 +34,115 @@ std::vector<uint8_t> _PyOpcode_Caches_vec;
 
 using torch::dynamo::autograd::torch_c_dynamo_compiled_autograd_init;
 
+namespace {
+
+template <typename T>
+bool unicode_is_literal_none(const T* start, const T* end) {
+  if (end != start + 4) {
+    return false;
+  }
+
+  return start[0] == 'N' && start[1] == 'o' && start[2] == 'n' &&
+      start[3] == 'e';
+}
+
+template <typename T>
+THPObjectPtr strip_function_call_helper(
+    PyObject* original,
+    const T* const start,
+    size_t length) {
+  // This function is... not great.
+  const T* const end = start + length;
+  const T* curr = start;
+  for (auto p = start; p < end; ++p) {
+    if (*p == ' ' || *p == '(') {
+      curr = p + 1;
+    } else if (*p == ')' || *p == ',' || *p == '[' || *p == ']') {
+      if ((p > curr) && !unicode_is_literal_none(curr, p) &&
+          (Py_UNICODE_ISALPHA(*curr) || *curr == '_')) {
+        return strip_function_call_helper(nullptr, curr, p - curr);
+      }
+      // The original code skipped adding these chars...
+    }
+  }
+
+  // strip_getattr_getitem
+  auto p = start;
+  for (; p < end; ++p) {
+    if (*p == '.' || *p == '[')
+      break;
+  }
+
+  if (p == end && original) {
+    return THPObjectPtr::dup(original);
+  }
+
+  return THPObjectPtr(
+      PyUnicode_FromKindAndData(sizeof(*start), start, p - start));
+}
+
+THPObjectPtr strip_function_call(PyObject* name) {
+  if (!PyUnicode_Check(name)) {
+    PyErr_SetString(PyExc_TypeError, "String expected");
+    return THPObjectPtr::none();
```

Comment:
You can't return `None` if you are setting an exception. If you are using the Python C API directly, you must return `NULL` when setting an exception.

(Or use pybind11, or PyTorch's error handling: https://github.com/pytorch/pytorch/blob/main/torch/csrc/README.md#exceptions)